src/moscot/problems/generic/_mixins.py

```diff
@@ -86,13 +89,124 @@ def cell_transition(
             key_added=key_added,
         )
 
+    @d_mixins.dedent
+    def push(
+        self: GenericAnalysisMixinProtocol[K, B],
+        source: K,
+        target: K,
+        data: Optional[Union[str, ArrayLike]] = None,
+        subset: Optional[Union[str, List[str], Tuple[int, int]]] = None,
+        scale_by_marginals: bool = True,
+        key_added: Optional[str] = PlottingDefaults.PUSH,
+        return_all: bool = False,
+        return_data: bool = False,
+        **kwargs: Any,
+    ) -> Optional[ApplyOutput_t[K]]:
+        """
+        Push distribution of cells through time.
+
+        Parameters
+        ----------
+        %(source)s
+        %(target)s
+        %(data)s
+        %(subset)s
+        %(scale_by_marginals)s
+        %(key_added_plotting)s
+        %(return_all)s
+        %(return_data)s
+
+        Return
+        ------
+        %(return_push_pull)s
+
+        """
+        result = self._apply(
+            start=source,
+            end=target,
+            data=data,
+            subset=subset,
+            forward=True,
+            return_all=return_all or key_added is not None,
+            scale_by_marginals=scale_by_marginals,
+            **kwargs,
+        )
+
+        if TYPE_CHECKING:
+            assert isinstance(result, dict)
+
+        if key_added is not None:
+            if TYPE_CHECKING:
+                assert isinstance(key_added, str)
+            plot_vars = {
+                "distribution_key": self.batch_key,
+            }
+            self.adata.obs[key_added] = self._flatten(result, key=self.batch_key)
+            Key.uns.set_plotting_vars(self.adata, AdataKeys.UNS, PlottingKeys.PUSH, key_added, plot_vars)
```

Comment:
Why does `set_plotting_vars` need the `AdataKeys.UNS`?

Reply:
this is `moscot_results`, we save everything there. Or what do you mean?

Reply:
Since you call `Key.uns`, it seems like double info is used (i.e., it should always use `AdataKeys.UNS`), right? Would then remove this argument, but maybe better for another PR.